src/main/java/net/minecraftforge/client/sounds/generators/SoundDefinitionsProvider.java

```diff
@@ -0,0 +1,245 @@
+     */
+    protected void add(final ResourceLocation soundEvent, final SoundDefinition definition) {
+        this.addSounds(soundEvent.getPath(), definition);
+    }
+
+    /**
+     * Adds the {@link SoundEvent} with the specified name along with its {@link SoundDefinition}
+     * to the list.
+     *
+     * <p>The given sound event must NOT contain the namespace the name is a part of, since
+     * the sound definition specification doesn't allow sounds to be defined outside the
+     * namespace they're in. For this reason, any namespace will automatically be stripped
+     * from the name.</p>
+     *
+     * @param soundEvent The name of the {@link SoundEvent}.
+     * @param definition The {@link SoundDefinition} that defines the given event.
+     */
+    protected void add(final String soundEvent, final SoundDefinition definition) {
+        this.add(new ResourceLocation(soundEvent), definition);
+    }
+
+    private void addSounds(final String soundEvent, final SoundDefinition definition) {
+        if (this.sounds.put(soundEvent, definition) != null) {
+            throw new IllegalStateException("Sound event '" + this.modId + ":" + soundEvent + "' already exists");
+        }
+    }
+
+    // Internal handling stuff
+    private void validate() {
+        final List<String> notValid = this.sounds.entrySet().stream()
+                .filter(it -> this.validate(it.getKey(), it.getValue()))
+                .map(Map.Entry::getKey)
+                .map(it -> this.modId + ":" + it)
+                .collect(Collectors.toList());
+        if (!notValid.isEmpty()) {
+            throw new IllegalStateException("Found invalid sound events: " + notValid);
+        }
+    }
+
+    private boolean validate(final String name, final SoundDefinition def) {
+        return def.soundList().stream().anyMatch(it -> this.validate(name,it));
+    }
+
+    private boolean validate(final String name, final SoundDefinition.Sound sound) {
+        switch (sound.type()) {
+            case SOUND: return this.validateSound(name, sound.name());
+            case EVENT: return this.validateEvent(name, sound.name());
+        }
+        throw new AssertionError();
+    }
+
+    private boolean validateSound(final String soundName, final ResourceLocation name) {
+        final boolean valid = this.helper.exists(name, ResourcePackType.CLIENT_RESOURCES, ".ogg", "sounds");
+        if (!valid) {
+            final String path = name.getNamespace() + ":sounds/" + name.getPath() + ".ogg";
+            if (!"minecraft".equals(name.getNamespace())) {
+                LOGGER.warn("Unable to find corresponding OGG file '{}' for sound event '{}'", path, soundName);
+            } else {
+                LOGGER.warn("Unable to find original Minecraft sound OGG file '{}' for event '{}': assuming it's part of the data pack downloaded by the launcher", path, soundName);
+                return false;
```

Comment:
So a missing minecraft sound event is marked as not valid.
Did you not specify in your PR message that it lets this slide cause it might be missing due to ExternalFileHelper not seeing those?

Reply:
Okey after a discussion on discord, this message can be disregarded I misread the methods intentions and return values.
A bit weirdly named, to be fair. But this works as intended.

Reply:
After discussion on Discord, I decided to flip the logic so that it is easier to understand. Double negatives aren't that good.